audio-analysis/main.py

```diff
@@ -39,39 +42,55 @@ def analyzeVideoSound(url, tag):
         else:
             print('Join failed ... each word does not have an amplitude')
         print(df3)
+        #Export video breakdown to CSV
         df3.to_csv("data_export.csv", header=True)
-
-        engine = create_engine('mysql+pymysql://root:root@localhost/climactic_test')
-        df3.to_sql("test_table", con=engine, if_exists='replace')
-
-
-    
+        #Export video breakdown to SQL
+        df3.to_sql("test_table", con=engine, if_exists='append')
 
 ##### Prompt for Highlight Video and Tagging #####
-answer = input("Option 1. single | Option 2. csv | Option 3. test \n")
+def prompt():
+    answer = input("Option 1. single | Option 2. csv | Option 3. test \n")
 
-#If user wants to upload only a single video
-if answer == 'single' or answer == '1':
-    url = input("Provide YT Link: ")
-    tag = input("Provide highlight tag (True/False): ")
-    print("Link: ", url)
-    print("Tag: ", tag)
-    analyzeVideoSound(url, tag)
+    #If user wants to upload only a single video
+    if answer == 'single' or answer == '1':
+        url = input("Provide YT Link: ")
+        tag = input("Provide highlight tag (True/False): ")
+        print("Link: ", url)
+        print("Tag: ", tag)
+        #Check if video already exists
+        url_exists = urlExists(url)
+        if url_exists == True:
+            print("This video is already in the database! Did not export.")
+        else:
+            analyzeVideoSound(url, tag)
+
+    #If user wants to upload only a CSV of videos
+    elif answer == 'csv' or answer == '2':
+        inputCSV = input("Provide csv with file and path \n")
+        tuplesDF = pd.read_csv(inputCSV)
+        print(tuplesDF.head())
+        i = 0
+        while i < len(tuplesDF.index):
+            print("\n For Video #", i+1, ":")
+            url = tuplesDF.loc[i, 'YT_LINK']
+            tag = tuplesDF.loc[i, 'H_TAG']
+            print("Link: ", url, " | Tag: ", tag)
+            #Check if video already exists
+            url_exists = urlExists(url)
+            if url_exists == True:
+                print("This video is already in the database! Did not export.")
+            else:
+                analyzeVideoSound(url, tag)
+            i += 1
+            print("\n --------------------------------")
 
-#If user wants to upload only a CSV of videos
-elif answer == 'csv' or answer == '2':
-    inputCSV = input("Provide csv with file and path \n")
-    tuplesDF = pd.read_csv(inputCSV)
-    print(tuplesDF.head())
-    i = 0
-    while i < len(tuplesDF.index):
-        print("\n For Video #", i+1, ":")
-        url = tuplesDF.loc[i, 'YT_LINK']
-        tag = tuplesDF.loc[i, 'H_TAG']
-        print("Link: ", url, " | Tag: ", tag)
-        analyzeVideoSound(url, tag)
-        i += 1
-        print("\n --------------------------------")
+    elif answer == 'test' or answer == '3':
+        #Check if video already exists
+        url_exists = urlExists("https://www.youtube.com/watch?v=JZRXESV3R74")
```

Comment:
Maybe we can bring out  `https://www.youtube.com/watch?v=JZRXESV3R74` into a constant

Reply:
Fixed